pkg/validation/eval.go

```diff
@@ -172,15 +185,23 @@ func (ev *Evaluator) Evaluate(rules []Rule, vm *k6tv1.VirtualMachine) *Result {
 		// we simply skip the malformed rule, the error can go unnoticed.
 		// IOW, this is a policy decision
 		if ok, err := ev.isRuleWellFormed(r, names); !ok {
-			result.Fail(r, err)
+			if r.JustWarning {
+				result.Warn(r.Message, err)
+			} else {
+				result.Fail(r, err)
+			}
 			continue
 		}
 
 		// Specialize() may be costly, so we do this before.
 		ok, err := r.IsAppliableOn(vm)
 		if err != nil {
 			fmt.Fprintf(ev.Sink, "%s failed: not appliable: %v\n", r.Name, err)
-			result.Fail(r, err)
+			if r.JustWarning {
```

Comment:
Here the change looks correct. Here we are detecting syntactically (aka well formed) rules that somehow don't make sense in the given VM.